dac: remove Kconfig.defconfig setting of DAC drivers

galak opened the pull request:
Now that DAC drivers are enabled based on devicetree
we need to remove any cases of them getting enabled by
Kconfig.defconfig* files as this can lead to errors.

Typically the Kconfig.defconfig* will blindly enable a
sensor and not respect the devicetree state of the DAC.
Additionally we can get problems with prj.conf/defconfig
getting incorrectly overridden.

Signed-off-by: Kumar Gala <galak@kernel.org>

galak committed:
dac: remove Kconfig.defconfig setting of DAC drivers

Now that DAC drivers are enabled based on devicetree
we need to remove any cases of them getting enabled by
Kconfig.defconfig* files as this can lead to errors.

Typically the Kconfig.defconfig* will blindly enable a
sensor and not respect the devicetree state of the DAC.
Additionally we can get problems with prj.conf/defconfig
getting incorrectly overridden.

Signed-off-by: Kumar Gala <galak@kernel.org>

martinjaeger commented on boards/arm/bl653_dvk/Kconfig.defconfig:
```diff
@@ -17,9 +17,6 @@ config BT_CTLR
 
 if DAC
 
-config DAC_MCP4725
```
I think you can remove the entire `if DAC [...] endif` section (here and for some other boards). I2C should also be enabled via devicetree and `DAC_INIT_PRIORITY` is now obsolete for this driver, as it uses its own init priority, see [here](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/dac/Kconfig.mcp4725).

galak replied:
So I2C bus is NOT enabled by devicetree.

galak replied:
I added a commit to remove the DAC_INIT_PRIORITY.  However I think the I2C and SPI bus bits is still needed.

galak committed:
dac: remove setting DAC_INIT_PRIORITY in board Kconfig.defconfig

The specific drivers for DAC_MCP4725 and DAC_DACX0508 have their
own driver specific Kconfig symbol for the INIT_PRIORITY.  So
remove setting the generic one as it doesnt make sense to set it
and the default value matches what the board specific values for
INIT_PRIORITY are set to.

Signed-off-by: Kumar Gala <galak@kernel.org>
